src/modules/token/index.ts

```diff
@@ -27,6 +31,16 @@ export interface Erc20Token {
   isCbridgeToken?: boolean;
 }
 
+export interface XvmAsset {
```

Comment:
I'm kinda wondering about the overall code structure here. Why do we need a separate `XvmAsset` if the interface and usage are the same as `Erc20Token` (though if `XvmAsset` was inheriting from `Erc20Token`, it makes sense)?

And when I looked around the code, you added a bunch of XVM-specific components and logic.

This will become an issue when we later decide to support PSP22 and the assets pallet like we support ERC20. I would recommend that the interface for XVM-specific calls should be at a minimum, and try to imagine if we are adding support for Assets pallet and PSP22 tokens at the same time.

Reply:
> I'm kinda wondering about the overall code structure here. Why do we need a separate XvmAsset if the interface and usage are the same as Erc20Token (though if XvmAsset was inheriting from Erc20Token, it makes sense)?

You are right, I've changed to reuse `ERC20Token` interface instead of creating new interface.

commit:
https://github.com/AstarNetwork/astar-apps/pull/627/commits/2451cc4f3b61c81d4eec388e024cc3b7934e8902

Reply:
> And when I looked around the code, you added a bunch of XVM-specific components and logic.
This will become an issue when we later decide to support PSP22 and the assets pallet like we support ERC20. I would recommend that the interface for XVM-specific calls should be at a minimum, and try to imagine if we are adding support for Assets pallet and PSP22 tokens at the same time.

I see your point and I agree with you. Actually, I always try to reduce the amount of logic, but it is also difficult for me to consider about supporting `PSP22` on our potral at this moment. I will be happy to refactor it when we are ready to implement `PSP22` tokens.